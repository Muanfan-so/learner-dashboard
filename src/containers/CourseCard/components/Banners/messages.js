import { StrictDict } from 'utils';

export const messages = StrictDict({
  auditAccessExpired: {
    id: 'learner-dash.courseCard.banners.auditAccessExpired',
    description: 'Audit access expiration banner message',
    defaultMessage: 'Your audit access to this course has expired.',
  },
  upgradeToAccess: {
    id: 'learner-dash.courseCard.banners.upgradeToAccess',
    description: 'Upgrade prompt for audit-expired learners that can still upgrade',
    defaultMessage: 'Upgrade now to access your course again.',
  },
  findAnotherCourse: {
    id: 'learner-dash.courseCard.banners.findAnotherCourse',
    description: 'Action prompt taking learners to course exploration',
    defaultMessage: 'Find another course',
  },
  upgradeDeadlinePassed: {
    id: 'learner-dash.courseCard.banners.upgradeDeadlinePassed',
    description: 'Audit upgrade deadline passed banner message',
    defaultMessage: 'Your upgrade deadline for this course has passed.  To upgrade, enroll in a session that is farther in the future.',
  },
  exploreCourseDetails: {
    id: 'learner-dash.courseCard.banners.exploreCourseDetails',
    description: 'Action prompt taking learners to course details page',
    defaultMessage: 'Explore course details.',
  },
  certRestricted: {
    id: 'learner-dash.courseCard.banners.certificateRestricted',
    description: 'Restricted certificate warning message',
    defaultMessage: 'Your Certificate of Achievement is being held pending confirmation that the issuance of your Certificate is in compliance with strict U.S. embargoes on Iran, Cuba, Syria, and Sudan.  If you think our system has mistakenly identified you as being connected with one of those countries, please let us know by contacting {supportEmail}.',
  },
  certRefundContactBilling: {
    id: 'learner-dash.courseCard.banners.certificateRefundContactBilling',
    description: 'Message to learners to contact billing for certificate refunds',
    defaultMessage: 'If you would like a refund on your Certificate of Achievement, please contact our billing address {billingEmail}',
  },
  passingGrade: {
    id: 'learner-dash.courseCard.banners.passingGrade',
    description: 'Message to learners with minimum passing grade for the course',
    defaultMessage: 'Grade required to pass the course: {minPassingGrade}',
  },
  notEligibleForCert: {
    id: 'learner-dash.courseCard.banners.notEligibleForCert',
    description: 'Certificate inelligibility message',
    defaultMessage: 'You are not eligible for a certificate.',
  },
  viewGrades: {
    id: 'learner-dash.courseCard.banners.viewGrades',
    description: 'Gradses link text',
    defaultMessage: 'View grades.',
  },
  certReady: {
    id: 'learner-dash.courseCard.banners.certReady',
    description: 'Certificate ready message',
    defaultMessage: 'Congratulations.  Your certificate is ready.',
  },
  viewCertificate: {
    id: 'learner-dash.courseCard.banners.viewCertificate',
    description: 'Certificate link text',
    defaultMessage: 'View Certificate.',
  },
  certMinGrade: {
    id: 'learner-dash.courseCard.banners.certMinGrade',
    description: 'Passing grade requirement message',
    defaultMessage: 'Grade required for a certificate: {minPassingGrade}',
  },
  downloadCertificate: {
    id: 'learner-dash.courseCard.banners.downloadCertificate',
    description: 'Certificate download link text',
    defaultMessage: 'Download Certificate.',
  },
  gradeAndCertReadyAfter: {
    id: 'learner-dash.courseCard.banners.gradseAndCertReadyAfter',
    description: 'Grade and certificate availability date message',
    defaultMessage: 'Your grade and certificate will be ready after {availableDate}.',
  },
  entitlementsUnavailable: {
    id: 'learner-dash.courseCard.banners.entitlementsUnavailable',
    description: 'Entitlements course message when no sessions are available',
    defaultMessage: 'There are no sessions available at the moment.  The course team will create new sessions soon.  If no sessions appear, please contact {emailLink} for information.',
  },
  entitlementsExpiringSoon: {
    id: 'learner-dash.courseCard.banners.entitlementsExpiringSoon',
    description: 'Entitlements course message when the entitlement is expiring soon.',
    defaultMessage: 'You must {selectSessionButton} by {changeDeadline} to access the course.',
  },
  entitlementsExpired: {
    id: 'learner-dash.courseCard.banners.entitlementsExpired',
    description: 'Entitlements course message when the entitlement is expired.',
    defaultMessage: 'You can no longer change sessions.',
  },
  selectSession: {
    id: 'learner-dash.courseCard.banners.selectSession',
    description: 'Entitlements session selection link text',
    defaultMessage: 'select a session',
  },
});

export default messages;
