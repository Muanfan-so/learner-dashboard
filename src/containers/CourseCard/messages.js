import { StrictDict } from 'utils';

export const messages = StrictDict({
  bannerAlt: {
    id: 'learner-dash.courseCard.bannerAlt',
    description: 'Course card banner alt-text',
    defaultMessage: 'Course thumbnail',
  },
});

export default messages;
