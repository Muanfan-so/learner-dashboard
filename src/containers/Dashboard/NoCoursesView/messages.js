import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  lookingForChallengePrompt: {
    id: 'EmptyCourse.lookingForChallengePrompt',
    defaultMessage: 'Looking for a new challenge?',
    description: 'Prompt user for new challenge',
  },
  exploreCoursesPrompt: {
    id: 'EmptyCourse.exploreCoursesPrompt',
    defaultMessage: 'Explore our courses to add them to your dashboard.',
    description: 'Prompt user to explore more courses',
  },
  exploreCoursesButton: {
    id: 'EmptyCourse.exploreCoursesButton',
    defaultMessage: 'Explore courses',
    description: 'Button to explore more courses',
  },
});

export default messages;
