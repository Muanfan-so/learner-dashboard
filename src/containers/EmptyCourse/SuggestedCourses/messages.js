import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  header: {
    id: 'leanerDashboard.emptyCourse.suggestedCourses.header',
    defaultMessage: 'Some courses you may be interested in',
    description: 'Header for suggesting courses',
  },
  viewCourseButton: {
    id: 'leanerDashboard.emptyCourse.suggestedCourses.viewCourseButton',
    defaultMessage: 'View Course',
    description: 'Actions button to visit the suggested course',
  },
  courseImageAlt: {
    id: 'leanerDashboard.emptyCourse.suggestedCourses.courseImageAlt',
    defaultMessage: 'Course image banner',
    description: 'alt for course image',
  },
  institueLogoAlt: {
    id: 'leanerDashboard.emptyCourse.suggestedCourses.institueLogoAlt',
    defaultMessage: 'Course institute logo',
    description: 'alt for institute logo',
  },
});

export default messages;
