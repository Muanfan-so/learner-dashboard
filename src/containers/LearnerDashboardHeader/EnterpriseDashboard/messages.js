import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  dashboard: {
    id: 'leanerDashboard.menu.dashboard.label',
    defaultMessage: 'Dashboard',
    description: 'The text for the user menu Dashboard navigation link.',
  },
  enterpriseDialogHeader: {
    id: 'leanerDashboard.enterpriseDialogHeader',
    defaultMessage: 'You have access to the {label} dashboard',
    description: 'title for enterpise dashboard dialog',
  },
  enterpriseDialogBody: {
    id: 'leanerDashboard.enterpriseDialogBody',
    defaultMessage: 'To access the coureses available to you through {label}, visit the {label} dashboard now.',
    description: 'Body text for enterpise dashboard dialog',
  },
  enterpriseDialogDismissButton: {
    id: 'leanerDashboard.enterpriseDialogDismissButton',
    defaultMessage: 'Dismiss',
    description: 'Dismiss button to cancel visiting dashboard',
  },
  enterpriseDialogConfirmButton: {
    id: 'leanerDashboard.enterpriseDialogConfirmButton',
    defaultMessage: 'Go To Dashboard',
    description: 'Confirm button to go to the dashboard url',
  },
});

export default messages;
