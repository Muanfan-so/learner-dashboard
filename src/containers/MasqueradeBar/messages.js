import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  ViewAs: {
    id: 'MasqueradeBar.ViewAs',
    defaultMessage: 'View as: ',
    description: 'Label for the View as',
  },
  ViewingAs: {
    id: 'MasqueradeBar.ViewingAs',
    defaultMessage: 'Viewing as: ',
    description: 'Label for the Viewing as',
  },
  SubmitButton: {
    id: 'MasqueradeBar.SubmitButton',
    defaultMessage: 'Submit',
    description: 'Label for the Submit button',
  },
  StudentNameInput: {
    id: 'MasqueradeBar.StudentNameInput',
    defaultMessage: 'Student username or email',
    description: 'Label for the Student Name or email input',
  },
});

export default messages;
