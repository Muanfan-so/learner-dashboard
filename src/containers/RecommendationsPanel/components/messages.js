import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  courseTitle: {
    id: 'course.title',
    defaultMessage: '{courseTitle}',
    description: 'Course title',
  },
});

export default messages;
