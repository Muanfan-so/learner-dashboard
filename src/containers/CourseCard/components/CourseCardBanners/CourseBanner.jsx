/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import { Hyperlink } from '@edx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';

import { hooks as appHooks } from 'data/redux';
import Banner from 'components/Banner';
import messages from './messages';

export const CourseBanner = ({ cardId }) => {
  const {
    isVerified,
    isAuditAccessExpired,
    canUpgrade,
    coursewareAccess = {},
  } = appHooks.useCardEnrollmentData(cardId);
  const courseRun = appHooks.useCardCourseRunData(cardId);
  const course = appHooks.useCardCourseData(cardId);
  const { formatMessage } = useIntl();

  const {
    hasUnmetPrerequisites,
    isStaff,
    isTooEarly,
  } = coursewareAccess;

  const bannerStackComponents = [];

  if (isVerified) { return null; }

  if (isAuditAccessExpired) {
    if (canUpgrade) {
      bannerStackComponents.push(
        <Banner>
          {formatMessage(messages.auditAccessExpired)}
          {'  '}
          {formatMessage(messages.upgradeToAccess)}
        </Banner>,
      );
    }
    bannerStackComponents.push(
      <Banner>
        {formatMessage(messages.auditAccessExpired)}
        {'  '}
        <Hyperlink destination="">{formatMessage(messages.findAnotherCourse)}</Hyperlink>
      </Banner>,
    );
  }
  if (courseRun.isActive && !canUpgrade) {
    bannerStackComponents.push(
      <Banner>
        {formatMessage(messages.upgradeDeadlinePassed)}
        {'  '}
        <Hyperlink destination={course.website || ''}>
          {formatMessage(messages.exploreCourseDetails)}
        </Hyperlink>
      </Banner>,
    );
  }
  if (isTooEarly) {
    bannerStackComponents.push(
      <Banner>
        {formatMessage(messages.courseHasNotStarted, { startDate: courseRun.startDate })}
      </Banner>,
    );
  }
  if (hasUnmetPrerequisites) {
    bannerStackComponents.push(
      <Banner>
        {formatMessage(messages.prerequisitesNotMet)}
      </Banner>,
    );
  }
  if (isStaff) {
    bannerStackComponents.push(
      <Banner>
        {formatMessage(messages.staffAccessOnly)}
      </Banner>,
    );
  }
  return <>{bannerStackComponents}</>;
};
CourseBanner.propTypes = {
  cardId: PropTypes.string.isRequired,
};

export default CourseBanner;
