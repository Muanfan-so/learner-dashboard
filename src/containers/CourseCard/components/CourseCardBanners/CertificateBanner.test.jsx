import React from 'react';
import { shallow } from 'enzyme';
import { Hyperlink } from '@edx/paragon';

import { hooks as appHooks } from 'data/redux';
import { CourseBanner } from './CourseBanner';

import messages from './messages';

jest.mock('components/Banner', () => 'Banner');
jest.mock('data/redux', () => ({
  hooks: {
    useCardCourseData: jest.fn(),
    useCardCourseRunData: jest.fn(),
    useCardEnrollmentData: jest.fn(),
  },
}));

const cardId = 'my-test-course-number';

let el;

const enrollmentData = {
  isVerified: false,
  canUpgrade: false,
  isAuditAccessExpired: false,
};
const courseRunData = {
  isActive: false,
};
const courseData = {
  website: 'test-course-website',
};

const render = (overrides = {}) => {
  const {
    course = {},
    courseRun = {},
    enrollment = {},
  } = overrides;
  appHooks.useCardCourseData.mockReturnValueOnce({
    ...courseData,
    ...course,
  });
  appHooks.useCardCourseRunData.mockReturnValueOnce({
    ...courseRunData,
    ...courseRun,
  });
  appHooks.useCardEnrollmentData.mockReturnValueOnce({
    ...enrollmentData,
    ...enrollment,
  });
  el = shallow(<CourseBanner cardId={cardId} />);
};

describe('CourseBanner', () => {
  it('initializes data with course number from enrollment, course and course run data', () => {
    render();
    expect(appHooks.useCardCourseData).toHaveBeenCalledWith(cardId);
    expect(appHooks.useCardCourseRunData).toHaveBeenCalledWith(cardId);
    expect(appHooks.useCardEnrollmentData).toHaveBeenCalledWith(cardId);
  });
  test('no display if learner is verified', () => {
    render({ enrollment: { isVerified: true } });
    expect(el.isEmptyRender()).toEqual(true);
  });
  describe('audit access expired, can upgrade', () => {
    beforeEach(() => {
      render({ enrollment: { isAuditAccessExpired: true, canUpgrade: true } });
    });
    test('snapshot: (auditAccessExpired, upgradeToAccess)', () => {
      expect(el).toMatchSnapshot();
    });
    test('messages: (auditAccessExpired, upgradeToAccess)', () => {
      expect(el.text()).toContain(messages.auditAccessExpired.defaultMessage);
      expect(el.text()).toContain(messages.upgradeToAccess.defaultMessage);
    });
  });
  describe('audit access expired, cannot upgrade', () => {
    beforeEach(() => {
      render({ enrollment: { isAuditAccessExpired: true } });
    });
    test('snapshot: (auditAccessExpired, findAnotherCourse hyperlink)', () => {
      expect(el).toMatchSnapshot();
    });
    test('messages: (auditAccessExpired, upgradeToAccess)', () => {
      expect(el.text()).toContain(messages.auditAccessExpired.defaultMessage);
      expect(el.find(Hyperlink).text()).toEqual(messages.findAnotherCourse.defaultMessage);
    });
  });
  describe('course run active and cannot upgrade', () => {
    beforeEach(() => {
      render({ courseRun: { isActive: true } });
    });
    test('snapshot: (upgradseDeadlinePassed, exploreCourseDetails hyperlink)', () => {
      expect(el).toMatchSnapshot();
    });
    test('messages: (upgradseDeadlinePassed, exploreCourseDetails hyperlink)', () => {
      expect(el.text()).toContain(messages.upgradeDeadlinePassed.defaultMessage);
      const link = el.find(Hyperlink);
      expect(link.text()).toEqual(messages.exploreCourseDetails.defaultMessage);
      expect(link.props().destination).toEqual(courseData.website);
    });
  });
  test('no display if audit access not expired and (course is not active or can upgrade)', () => {
    render();
    expect(el.html()).toEqual('');
    render({ enrollment: { canUpgrade: true }, courseRun: { isActive: true } });
    expect(el.html()).toEqual('');
  });
});
