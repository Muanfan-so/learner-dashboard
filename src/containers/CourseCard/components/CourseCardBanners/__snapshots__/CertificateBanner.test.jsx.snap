// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseBanner audit access expired, can upgrade snapshot: (auditAccessExpired, upgradeToAccess) 1`] = `
<Fragment>
  <Banner>
    Your audit access to this course has expired.
      
    Upgrade now to access your course again.
  </Banner>
</Fragment>
`;

exports[`CourseBanner audit access expired, cannot upgrade snapshot: (auditAccessExpired, findAnotherCourse hyperlink) 1`] = `
<Fragment>
  <Banner>
    Your audit access to this course has expired.
      
    <Hyperlink
      destination=""
    >
      Find another course
    </Hyperlink>
  </Banner>
</Fragment>
`;

exports[`CourseBanner course run active and cannot upgrade snapshot: (upgradseDeadlinePassed, exploreCourseDetails hyperlink) 1`] = `
<Fragment>
  <Banner>
    Your upgrade deadline for this course has passed.  To upgrade, enroll in a session that is farther in the future.
      
    <Hyperlink
      destination="test-course-website"
    >
      Explore course details.
    </Hyperlink>
  </Banner>
</Fragment>
`;
